stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/attachments/ImageAttachmentFactory.kt

```diff
@@ -0,0 +1,168 @@
+package io.getstream.chat.android.compose.ui.attachments
+
+import androidx.compose.foundation.ExperimentalFoundationApi
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.combinedClickable
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.wrapContentSize
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import coil.compose.rememberImagePainter
+import com.getstream.sdk.chat.utils.extensions.imagePreviewUrl
+import io.getstream.chat.android.client.models.Message
+import io.getstream.chat.android.compose.R
+import io.getstream.chat.android.compose.state.messages.attachments.AttachmentState
+import io.getstream.chat.android.compose.ui.imagepreview.ImagePreviewActivity
+import io.getstream.chat.android.compose.ui.theme.ChatTheme
+
+/**
+ * An extension of the [AttachmentFactory] that validates attachments as images and uses [ImageAttachmentContent] to
+ * build the UI for the message.
+ * */
+public class ImageAttachmentFactory : AttachmentFactory() {
+
+    override fun predicate(message: Message): Boolean {
+        return message.attachments.all { it.type == "image" }
+    }
+
+    @Composable
+    public override fun Content(attachmentState: AttachmentState) {
+        ImageAttachmentContent(attachmentState)
+    }
+}
+
+/**
+ * Builds an image attachment message, which can be composed of several images.
+ *
+ * @param attachmentState - The state of the attachment, holding the root modifier, the message
+ * and the onLongItemClick handler.
+ * */
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+public fun ImageAttachmentContent(attachmentState: AttachmentState) {
+    val (modifier, messageItem, onLongItemClick) = attachmentState
+    val (message, _) = messageItem
+    val context = LocalContext.current
+
+    Row(
+        modifier
+            .size(height = 200.dp, width = 250.dp)
```

Comment:
+1 on this for customization, trying to understand the decision to add similar lines  for new components

Reply:
Similar to the comment above - this is according to our own default UI, so if people want to customize it more, we'll probably expose things like `messageWidth` or just let people customize the entire item in an attachment, or just override the factory.